datasets/chorus-dt/scripts/resolve_places.py

```diff
@@ -16,25 +16,37 @@
 tqdm.pandas()
 
 
-def compute_distance(x):
-    lon1 = x["coords_place_0_lon"]
-    lat1 = x["coords_place_0_lat"]
-    lon2 = x["coords_place_1_lon"]
-    lat2 = x["coords_place_1_lat"]
-    lon1, lat1, lon2, lat2 = map(radians, [lon1, lat1, lon2, lat2])
+def compute_distance(df, n):
+    
+    coord_lon_str = 'coords_place_'+str(n)+'_lon'
+    coord_lat_str = 'coords_place_'+str(n)+'_lat'
+    coord_lon_p_str = 'coords_place_'+str(n-1)+'_lon'
+    coord_lat_p_str = 'coords_place_'+str(n-1)+'_lat'
+    
+    lon0 = df[coord_lon_str]
+    lat0 = df[coord_lat_str]
+    lon1 = df[coord_lon_p_str]
+    lat1 = df[coord_lat_p_str]
+        
+    lon0 = np.deg2rad(lon0)
+    lon1 = np.deg2rad(lon1)
+    lat0 = np.deg2rad(lat0)
+    lat1 = np.deg2rad(lat1)
 
     # haversine formula
-    dlon = lon2 - lon1
-    dlat = lat2 - lat1
-    a = sin(dlat / 2) ** 2 + cos(lat1) * cos(lat2) * sin(dlon / 2) ** 2
-    c = 2 * asin(sqrt(a))
+    dlon = lon1 - lon0
+    dlat = lat1 - lat0
+    a = np.sin((lat1 - lat0)/2) ** 2 + np.cos(lat1) * np.cos(lat0) * np.sin((lon1 - lon0)/ 2)** 2
```

Comment:
You can keep `dlon` and `dlat` you just declared above.
```suggestion
    a = np.sin((dlat)/2) ** 2 + np.cos(lat1) * np.cos(lat0) * np.sin((dlon)/ 2)** 2
```